crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+                }
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to get LSP status: {}", e)).await;
+                }
+            }
+        });
+    }
+}
+
+/// a struct that checks and prompts the user to enable remote cache
+pub struct RemoteCacheDiagnostic(pub Arc<Mutex<CommandBase>>);
+impl RemoteCacheDiagnostic {
+    pub fn new(base: CommandBase) -> Self {
+        Self(Arc::new(Mutex::new(base)))
+    }
+}
+
+impl Diagnostic for RemoteCacheDiagnostic {
+    const NAME: &'static str = "vercel.auth";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        let base = self.0.clone();
+        tokio::task::spawn(async move {
+            let result = {
+                let base = base.lock().await;
+                base.config()
+                    .map(|c| (c.team_id().is_some(), c.team_slug().is_some()))
+            };
+
+            let Ok((has_team_id, has_team_slug)) = result else {
+                Self::failed(chan, "Malformed config file".to_string()).await;
+                return;
+            };
+
+            Self::started(&chan, "Remote Cache".to_string()).await;
+            Self::log_line(&chan, "Checking credentials".to_string()).await;
+
+            if has_team_id || has_team_slug {
+                Self::done(chan, "Remote Cache enabled".to_string()).await;
+                return;
+            }
+
+            let result = {
+                Self::log_line(&chan, "Linking to remote cache".to_string()).await;
+                let mut base = base.lock().await;
+                let Some((stopped, resume)) = Self::suspend(&chan).await else {
+                    // the sender (terminal) was shut, ignore
+                    return;
+                };
+                stopped.await.unwrap();
+                let link_res = link(&mut base, false, crate::cli::LinkTarget::RemoteCache).await;
+                resume.send(()).unwrap();
+                link_res
+            };
+
+            match result {
+                Ok(_) => {
+                    Self::log_line(&chan, "Linked".to_string()).await;
+                    Self::done(chan, "Remote Cache enabled".to_string()).await
+                }
+                Err(link::Error::NotLinking) => {
+                    Self::not_applicable(chan, "Remote Cache opted out".to_string()).await
+                }
+                Err(e) => {
+                    Self::failed(chan, format!("Failed to link: {}", e)).await;
+                }
+            }
+        });
+    }
+}
+
+pub struct UpdateDiagnostic;
+
+impl Diagnostic for UpdateDiagnostic {
+    const NAME: &'static str = "turbo.update";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        tokio::task::spawn(async move {
+            Self::started(&chan, "Update Turborepo to latest version".to_string()).await;
+            Self::log_line(&chan, "Checking for updates...".to_string()).await;
+            let version = check_for_updates(
```

Comment:
The blocking is probably just the time it takes to do the check since each lint dumps its entire logs before moving to the next one which I don't think will be improved. spawn_blocking should help on very constrained devices though so we'll do it anyways.